application.go

```diff
@@ -676,6 +677,11 @@ func (a *application) ProvisioningState() ProvisioningState {
 	return a.ProvisioningState_
 }
 
+// SetProvisioningState implements Application.
+func (a *application) SetProvisioningState(args *ProvisioningStateArgs) {
```

Comment:
Passing in a pointer here is a little inconvenient, since no other methods do

Reply:
It's because we don't want the zero value, we want to keep it nil if the arg is nil. This allows us to avoid the if !=nil check on the call sites.